alpaca/rest.go

```diff
@@ -703,6 +704,72 @@ func (c *Client) GetSnapshots(symbols []string) (map[string]*marketdata.Snapshot
 	return snapshots, nil
 }
 
+// GetADTV returns the average daily trade volume for a given symbol
+//
+// Deprecated: will be moved to the marketdata package!
+func (c *Client) GetADTV(symbol string, start, end time.Time) (*marketdata.ADTV, error) {
+	bars := c.GetBars(symbol, marketdata.Day, marketdata.Raw, start, end, 10000)
+	var (
+		count       int
+		totalVolume uint64
+	)
+	for barItem := range bars {
+		if barItem.Error != nil {
+			return nil, barItem.Error
+		}
+		count++
+		totalVolume += uint64(barItem.Bar.Volume)
+	}
+	if count == 0 {
+		return &marketdata.ADTV{Volume: 0}, nil
+	}
+	return &marketdata.ADTV{
+		Volume: float64(totalVolume) / float64(count),
+		Days:   count,
+	}, nil
+}
+
+// GetSMA returns the Simple Moving Average (SMA) starting at start, ending at end
+// and using window as the averaging period.
+//
+// Deprecated: will be moved to the marketdata package!
+func (c *Client) GetSMA(symbol string, start, end time.Time, window int) ([]marketdata.ValueAtTime, error) {
```

Comment:
This function has limited usage: eg. I could have SMA for volumes, opening/closing prices, or any other gauge data that I like. I could count all of them just to querying the underlying endpoint just once. If we implement everything passing all the information as parameters then the function signature will be too complex.